### PR TITLE
Remove a data node from the cluster

### PR DESCRIPTION
**Pull request by moesterheld**
## Description
Adds the REMOVING/REMOVED states and transitions to the datanode
image

Removal and Reset can be triggered from the Graylog REST API.
Graylog Server publishes events to the cluster event bus which are then consumed by the data nodes.

Next step is #17247 which is also needed to delete removed nodes from `unicast_hosts` upon restart.

## Motivation and Context
backend functionality for #16820 

## How Has This Been Tested?
- create data node cluster
- call DELETE on <graylog>/api/datanode/<nodeid> 
- when finished, POST to <graylog>/api/datanode/<nodeid>/reset

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by janheise** [comment]
@moesterheld if it makes sense, please add an IT test

File: data-node/src/main/java/org/graylog/datanode/rest/StatusController.java [hunk 1]
```diff
@@ -48,4 +49,9 @@ public DataNodeStatus status() {
                 new StatusResponse(datanodeConfiguration.opensearchDistributionProvider().get().version(), openSearch.processInfo())
         );
     }
+
+    @DELETE
+    public void remove() {
```
Comment (todvora) [inline data-node/src/main/java/org/graylog/datanode/rest/StatusController.java:54]: I'd suggest moving this to a different controller (management?), to keep the operations separated. 

The management controller probably shouldn't be accessible if the datanode is started in an insecure mode and there is no auth that would guard these, potentially very dangerous, actions.

**Review by todvora — COMMENTED** [review]
I think we need to secure the delete action or disable it if we are running in the insecure mode of the datanode.

File: data-node/src/main/java/org/graylog/datanode/rest/StatusController.java (hunk 1 above)
Comment (janheise) [reply data-node/src/main/java/org/graylog/datanode/rest/StatusController.java:54]: I agree

Comment (moesterheld) [reply data-node/src/main/java/org/graylog/datanode/rest/StatusController.java:54]: Good point. I only added the endpoint to be able to trigger removal from the integration test. 
I'll move it to a separate controller and secure it with an annotation

**Review by todvora — APPROVED** [review]
LGTM!

**Pull request merged**